### DISCUSSION TITLE
LESSON 7: The flow of the video doesn't make sense from 2:07 to 2:10

### DISCUSSION THREAD

**Discussion by mrthedude** [Cyfrin/foundry-full-course-cu#510] [General]
There's a section in the video where it seems the editing got mixed up. 

[2:07:34](https://youtu.be/sas02qSFZ74?t=7654)
Patrick deletes the ```vm.startBroadcast()``` and ```vm.stopBroadcast()``` from ```function fundFundMe()``` in file ```Interactions.s.sol``` to past them into the ```run()``` function but I don't understand why they're no longer necessary in the ```fundFundMe()``` function.
<br> 

At [2:07:54](https://youtu.be/sas02qSFZ74?t=7674) Patrick adds 5 lines of code: 
```
vm.prank(USER);
vm.deal(USER, 1e18);
fundFundMe.fundFundMe(address(fundMe));

address funder = fundMe.getFunder(0);
assertEq(funder, USER);
```
but the video cuts and the code is gone with no mention of what happened. 

<br><br>

At [2:09:13](https://youtu.be/sas02qSFZ74?t=7753), when I tested ```testUserCanFundInteractions()``` I kept getting the ```"EvmError: OutOfFund"``` error since the ```vm.startBroadcast()``` and ```vm.stopBroadcast()``` were still removed from the ```fundFundMe()``` function in file ```Interactions.s.sol```.

<br>

 At  [2:09:54](https://youtu.be/sas02qSFZ74?t=7794) Patrick scrolls through file ```Interactions.s.sol``` showing the ```vm.startBroadcast()``` and ```vm.stopBroadcast()``` back in ```fundFundMe()``` but there's no explanation.

It's just minor editing stuff but I wanted to mention it cause it through me for a loop the first time I watched it lol. If this is just something that I'm not seeing or understanding please let me know
<br>
With that being said I'm loving the course so far and have nothing but gratitude for the quality education you're making available!

**Comment by shikhar229169** [comment]
Yes, I was also stuck at that moment.
But the best part was debugging this made my concepts more clear.

**Reply by mrthedude** [reply]
100% agree with you there. I actually rewatched the entire lesson up to that point just to go over everything to see if i missed something

**Reply by LeeJianXing** [reply]
thanks for pointing that out, i was also super confuse at the duration of the video.

**Comment by SilviyaPBrayanova** [comment]
I too was confused. And I still am. Why calling fund() and withdraw() need to be surrounded with start/stop Broadcast() when called via script  function, but do not need to be if they are directly called by the test?
